examples/network_traffic_prediction/nr_models.py

```diff
@@ -0,0 +1,702 @@
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Demonstration models for using GraphStorm APIs
+"""
+
+import time
+import torch as th
+import graphstorm as gs
+from graphstorm.utils import barrier
+from graphstorm.model import (GSgnnNodeModel,
+                              GSNodeEncoderInputLayer,
+                              GSEdgeEncoderInputLayer,
+                              RelationalGCNEncoder,
+                              EntityRegression,
+                              RegressionLossFunc)
+from graphstorm.trainer import GSgnnNodePredictionTrainer
+from graphstorm.inference import GSgnnNodePredictionInferrer
+from graphstorm.model.utils import (append_to_dict,
+                                    save_shuffled_node_embeddings,
+                                    save_node_prediction_results,
+                                    NodeIDShuffler)
+
+
+def get_input_feat_size(feat_size, window_size, ts_feat_names, ts_size):
+    """ method sepcificly designed for the input feature size when having time series features
+    """
+    # reset feature size according to window size
+    new_efeat_size = {}
+    for a_type, feat_size in feat_size.items():
+        if a_type in ts_feat_names:
+            # new feature size is static + window_size time seria
+            new_efeat_size[a_type] = feat_size - len(ts_feat_names[a_type]) * ts_size  + window_size
+        else:
+            new_efeat_size[a_type] = feat_size
+
+    return new_efeat_size
+
+def get_st_feats(input_feats, ts_feat_names, ts_size):
+    """ Extract static node/edge features only
+
+    input_feats: dict of tensor, or list of dict of tensor
+        If a dict of tensor, it contains node features. If a list of features, it contains
+        edge features.
+    ts_feat_name: dict of string and list
+        A dict of node/edge type and time serias feature names in format {type, [str, ...])
+    ts_size: int
+        The overall time seria size, e.g., 24.
+
+    Tricks:
+    1: ts features are always be concatinated at the end of edge features in outputs.
+    2: all types of ts features have the same time seria size.
+
+    Return:
+    -------
+    new_st_feats: dict of tensor, or list of dict of tensor
+
+    """
+    if isinstance(input_feats, dict):
+        st_feats = {}
+        for a_type, feats in input_feats.items():
+            # has time seria features
+            if a_type in ts_feat_names:
+                ts_names = ts_feat_names[a_type]
+                num_ts_names = len(ts_names)
+                feats = feats[:, 0: -(num_ts_names * ts_size)]
```

Comment:
Because we always put time series features at the end of feature tensor, static feature will start from 0d to the time series feature dimension.

Added comments to explain this design.